pkg/pillar/docs/zedkube.md

```diff
@@ -0,0 +1,135 @@
+# Clustered eve nodes (zedkube)
+
+## Overview
+
+## Components
+
+### kubenodeop
+
+kubenodeop handles cordoning, uncordoning, and draining of clustered eve-os nodes.
+Any given node could be hosting one or more longhorn volume replicas and thus could be the rebuild source for other node replicas.
+A drain operation should be performed before any Node Operation / Node Command which can cause an extended outage of a node such as a reboot, shutdown, reset.
+kubenodeop handles NodeDrainRequest objects which zedkube subscribes to, initiates the drain, and publishes NodeDrainStatus objects.
+
+### Drain PubSub setup (node reboot/shutdown)
+
+1. zedagent/handlenodedrain.go:initNodeDrainPubSub()
+    - subscribes to NodeDrainStatus from zedkube
+    - creates publication of NodeDrainRequest
+1. nodeagent/handlenodedrain.go:initNodeDrainPubSub()
+    - subscribe to NodeDrainStatus from zedkube
+
+### Drain Request path (node reboot/shutdown)
+
+1. zedagent/parseconfig.go:scheduleDeviceOperation()
+    - Use GetNodeDrainStatus() to determine if system supports drain
+    - HV!=kubevirt: NOTSUPPORTED
+    - HV=kubevirt will return:
+        - NOTSUPPORTED if in single node.
+        - NOTREQUESTED if in cluster mode
+    - NodeDrainStatus == NOTREQUESTED:
+        - Drain is requested
+        - Reboot or shutdown cmd deferred state in zedagentContext
```

Comment:
Is this saying that this Get function will request a drain? 
If so I think the function should be named differently (e.g., RequestNodeDrain()) and it can return NOTSUPPORTED as an error.

Reply:
newer docs pushed up